test/test_core.py

```diff
@@ -1775,6 +1775,44 @@ def test_exceptions_longjmp3(self):
   def test_exceptions_longjmp4(self):
     self.do_core_test('test_exceptions_longjmp4.cpp')
 
+  def test_exception_sjlj_options(self):
+    # Emscripten EH and Wasm EH cannot be enabled at the same time
+    self.set_setting('DISABLE_EXCEPTION_CATCHING', 0)
+    err = self.expect_fail([EMCC, test_file('hello_world.cpp'), '-fwasm-exceptions'] + self.get_emcc_args())
+    self.assertContained('error: DISABLE_EXCEPTION_CATCHING=0 is not compatible with -fwasm-exceptions', err)
+    self.clear_all_settings()
```

Comment:
The problem with `clear_all_settings` is that it will also clear any settings that are part of the overall test suite.  For example the `lsan` and `msan` test suites set `ALLOW_MEMORY_GROWTH` .. and calling this would remove that setting half way through this test:

https://github.com/emscripten-core/emscripten/blob/0ced16e037d4d0dc43f9176db4fbdeb3b79d2df1/test/test_core.py#L9779

Maybe safer not to add this function at all?

Reply:
Removed `clear_all_settings`, and added `clear_all_relevant_settings` that removes only the relevant settings (`DISABLE_EXCEPTION_THROWING`, `DISABLE_EXCEPTION_CATCHING`, `SUPPORT_LONGJMP`, and `ASYNCIFY`) within the test.